File: app/src/androidTest/java/eu/sesma/dagger/platform/TestRunner.java
package eu.sesma.dagger.platform;

import android.app.Application;
import android.content.Context;
import android.support.test.runner.AndroidJUnitRunner;

public class TestRunner extends AndroidJUnitRunner {
    @Override
    public Application newApplication(ClassLoader cl, String className, Context context)
            throws InstantiationException, IllegalAccessException, ClassNotFoundException {
        return super.newApplication(cl, TestAndroidApplication.class.getName(), context);
    }
}

File: app/src/androidTest/java/eu/sesma/dagger/TestRunner.kt
package eu.sesma.dagger

import android.app.Application
import android.content.Context
import android.support.test.runner.AndroidJUnitRunner

class TestRunner : AndroidJUnitRunner() {

    @Throws(InstantiationException::class, IllegalAccessException::class, ClassNotFoundException::class)
    override fun newApplication(cl: ClassLoader, className: String, context: Context): Application {
        return super.newApplication(cl, TestAndroidApplication::class.java.name, context)
    }
}